csharp/src/Ice/LocatorInfo.cs

```diff
@@ -51,123 +59,90 @@ internal void ClearCache(Reference reference)
 
             if (reference.IsWellKnown)
             {
-                if (RemoveWellKnownProxy(reference) is Reference resolvedWellKnownProxy)
-                {
-                    if (resolvedWellKnownProxy.IsIndirect)
-                    {
-                        // We don't cache bogus well-known resolved references.
-                        Debug.Assert(!resolvedWellKnownProxy.IsWellKnown);
-
-                        if (reference.Communicator.TraceLevels.Locator >= 2)
-                        {
-                            TraceWellKnown("removed well-known proxy without endpoints from locator cache",
-                                           reference,
-                                           resolvedWellKnownProxy);
-                        }
-                        ClearCache(resolvedWellKnownProxy); // i.e clears location cache
-                    }
-                    else
-                    {
-                        if (reference.Communicator.TraceLevels.Locator >= 2)
-                        {
-                            TraceDirect("removed well-known proxy with endpoints from locator cache",
-                                        reference,
-                                        resolvedWellKnownProxy);
-                        }
-                    }
-                }
+                ClearCache(reference.Identity, reference.Protocol, reference.Communicator);
             }
             else
             {
-                if (RemoveLocation(reference) is Reference resolvedLocation &&
-                    reference.Communicator.TraceLevels.Locator >= 2)
-                {
-                    TraceDirect("removed endpoints for location from locator cache", reference, resolvedLocation);
-                }
+                ClearCache(reference.Location, reference.Protocol, reference.Communicator);
             }
         }
 
         /// <summary>Resolves an indirect reference using the locator proxy or cache.</summary>
-        internal async ValueTask<(Reference? DirectReference, bool Cached)> ResolveIndirectReferenceAsync(
+        internal async ValueTask<(EndpointList Endpoints, Location Location, bool Cached)>
+        ResolveIndirectReferenceAsync(
```

Comment:
Doesn't this messes up VSCode syntax highlighting for you as well when the return type and the method name are on separate lines? If that's the case, I think I'd rather keep everything on a single line even if it exceeds the number of 120 columns... or perhaps have an alias for such large return types.

Reply:
Maybe time to review our max line length, screens are getting bigger :)